src/codegen/printer.rs

```diff
@@ -314,6 +312,7 @@ impl std::fmt::Display for Ty {
                 BufferKind::Ident(name) => write!(f, "{}", name),
             },
             Scalar(sty) => write!(f, "{}", sty),
+            Atomic(at) => write!(f, "{}", at),
```

Comment:
If `Atomic` is used to generate `cuda::std::atomic` then this print is wrong. If an atomic type in descend should just be its scalar type in CUDA, then I think that `Atomic` should not be part of the CUDA AST.